sandbox.go

```diff
@@ -427,6 +428,8 @@ func (sb *sandbox) updateGateway(ep *endpoint) error {
 		return fmt.Errorf("failed to set IPv6 gateway while updating gateway: %v", err)
 	}
 
+	ep.network.addSvcRecords(ep.ID(), hostDockerInternal, ep.svcID, ep.Gateway(), ep.GatewayIPv6(), true, "updateGateway")
```

Comment:
I think you should wrap this under 
```
if (ep.needResolver()) {
}
```
since not endpoints participate in name resolution provided by the docker daemon (especially the default docker0 bridge case).